File: MPChartLib/src/main/java/com/github/mikephil/charting/formatter/IndexAxisValueFormatter.java
package com.github.mikephil.charting.formatter;

import java.util.Collection;

/**
 * This formatter is used for passing an array of x-axis labels, on whole x steps.
 */
public class IndexAxisValueFormatter extends ValueFormatter
{
    private String[] mValues = new String[] {};
    private int mValueCount = 0;

    /**
     * An empty constructor.
     * Use `setValues` to set the axis labels.
     */
    public IndexAxisValueFormatter() {
    }

    /**
     * Constructor that specifies axis labels.
     *
     * @param values The values string array
     */
    public IndexAxisValueFormatter(String[] values) {
        if (values != null)
            setValues(values);
    }

    /**
     * Constructor that specifies axis labels.
     *
     * @param values The values string array
     */
    public IndexAxisValueFormatter(Collection<String> values) {
        if (values != null)
            setValues(values.toArray(new String[values.size()]));
    }

    @Override
    public String getFormattedValue(float value) {
        int index = Math.round(value);

        if (index < 0 || index >= mValueCount || index != (int)value)
            return "";

        return mValues[index];
    }

    public String[] getValues()
    {
        return mValues;
    }

    public void setValues(String[] values)
    {
        if (values == null)
            values = new String[] {};

        this.mValues = values;
        this.mValueCount = values.length;
    }
}

File: MPChartLib/src/main/java/com/github/mikephil/charting/formatter/IndexAxisValueFormatter.kt
package com.github.mikephil.charting.formatter

import com.github.mikephil.charting.components.AxisBase
import kotlin.math.roundToInt

/**
 * This formatter is used for passing an array of x-axis labels, on whole x steps.
 */
open class IndexAxisValueFormatter : IAxisValueFormatter {

    var values: Array<String> = arrayOf()

    /**
     * Constructor that specifies axis labels.
     *
     * @param values The values string array
     */
    constructor(values: Array<String>?) {
        if (values != null)
            this.values = values
    }

    /**
     * Constructor that specifies axis labels.
     *
     * @param values The values string array
     */
    constructor(values: Collection<String>?) {
        if (values != null)
            this.values = values.toTypedArray()
    }

    override fun getFormattedValue(value: Float, axis: AxisBase?): String {
        val index = value.roundToInt()
        return if (index < 0 || index >= values.size || index != value.toInt())
            ""
        else
            values[index]
    }

}
